### PR TITLE
Options comparison document

### PR DESCRIPTION
**Pull request by psteinfeld**
This is the markdown version of the description of the command line options of several Fortran compilers.

### CODE REVIEW

File: documentation/OptionComparison.md [hunk 1]
```diff
@@ -0,0 +1,1332 @@
+<!--
+Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
```
Comment (klausler) [inline documentation/OptionComparison.md]: 2019

**Comment by psteinfeld** [comment]
I got rid of the hyphens to conserve space.  I had trouble fitting many of the options into the tables.

File: documentation/OptionComparison.md (hunk 1 above)
Comment (psteinfeld) [reply documentation/OptionComparison.md]: Thanks!  I fixed it.

File: documentation/OptionComparison.md [hunk 2]
```diff
@@ -0,0 +1,1332 @@
...
+   </td>
+   <td>N/A
+   </td>
+   <td>N/A
+   </td>
+   <td>N/A
+   </td>
+   <td><a href="https://software.intel.com/node/42e8ec08-64cc-44ba-a636-20ed50c682cd">altparam</a>
+   </td>
+   <td>N?A
```
Comment (tskeith) [inline documentation/OptionComparison.md]: N?A

File: documentation/OptionComparison.md [hunk 3]
```diff
@@ -0,0 +1,1332 @@
...
+*   **GNU:** The "std=_level_" option specifies the standard to which the program is expected to conform.   The default value for std is 'gnu', which specifies a superset of the latest Fortran standard that includes all of the extensions supported by GNU Fortran, although warnings will be given for obsolete extensions not recommended for use in new code. The 'legacy' value is equivalent but without the warnings for obsolete extensions. The 'f95', 'f2003', 'f2008', and 'f2018' values specify strict conformance to the respective standards.  Errors are given for all extensions beyond the relevant language standard, and warnings are given for the Fortran 77 features that are permitted but obsolescent in later standards. '-std=f2008ts' allows the Fortran 2008 standard including the additions of the Technical Specification (TS) 29113 on Further Interoperability of Fortran with C and TS 18508 on Additional Parallel Features in Fortran.  Values for "_level_" are f_95, f2003, f2008, f2008ts, f2018, gnu,_ and _legacy._
+
+**<a name="source"></a>Source format:** 
+
+**Fixed or free source: **  Cray, IBM, and Intel default the source format based on the source file suffix as follows:
```
Comment (tskeith) [inline documentation/OptionComparison.md]: The bolding doesn't seem to happen if there is a space before the final `**`.

File: documentation/OptionComparison.md [hunk 4]
```diff
@@ -0,0 +1,1332 @@
...
+
+
+
+*   **Cray**
+    *   **Free: **.f90, .F90, .f95, .F95, .f03, .F03, .f08, .F08, .ftn, .FTN
+    *   **Fixed:.**f, .F, .for, .FOR
```
Comment (tskeith) [inline documentation/OptionComparison.md]: Same problem here.

File: documentation/OptionComparison.md (hunk 3 above)
Comment (psteinfeld) [reply documentation/OptionComparison.md]: Thanks!  I fixed this.  GitHub handles this differently from Chrome.  Once I knew what to look for, I found many instances of this that needed to be fixed.

File: documentation/OptionComparison.md (hunk 4 above)
Comment (psteinfeld) [reply documentation/OptionComparison.md]: Yep.  One of many that I fixed.

File: documentation/OptionComparison.md [hunk 5]
```diff
@@ -0,0 +1,1332 @@
...
+*   **GNU:** For both "ffixed-line-length-_n_" and "ffree-line-length-_n_" options, characters are ignored after the specified length.  The default for fixed is 72.  The default for free is 132.  For free, you can specify 'none' as the length, which means that all characters in the line are meaningful.
+*   **IBM:** For **fixed**, the default is 72.  For **free**, there's no default, but the maximum length for either form is 132.
+*   **Intel:** The default is 72 for **fixed** and 132 for **free**.
+*   **PGI, Flang:** "-Mextend" instructs the compiler to accept 132-column source code; otherwise it accepts 72-column code.  It's unclear from the documentation whether this applies to fixed, free, or both source formats.
```
[the remark this answers is not in the document]
Comment (psteinfeld) [reply documentation/OptionComparison.md]: Thanks for the information!  I'll update the doc.

File: documentation/OptionComparison.md [hunk 6]
```diff
@@ -0,0 +1,1332 @@
...
+<p>
+<a href="https://software.intel.com/en-us/fortran-compiler-developer-guide-and-reference-assume">assume nostd_intent_in</a>,
+<p>
+<a href="https://software.intel.com/en-us/fortran-compiler-developer-guide-and-reference-assume">assume nostd_value</a>,
+<p>
+<a href="https://software.intel.com/en-us/fortran-compiler-developer-guide-and-reference-assume">assume norealloc_lhs</a>
+   </td>
+   <td>Mallocatable=95|03
+   </td>
+   <td>N/A
```
[the remark this answers is not in the document]
Comment (psteinfeld) [reply documentation/OptionComparison.md]: Thanks!  I'll update the doc.

File: documentation/OptionComparison.md [hunk 7]
```diff
@@ -0,0 +1,1332 @@
...
+   </td>
+   <td><a href="https://www-01.ibm.com/support/docview.wss?uid=swg27024803&aid=1#page=254">q recur</a>
+   </td>
+   <td><a href="https://software.intel.com/en-us/fortran-compiler-developer-guide-and-reference-assume">assume recursion</a>,
+<p>
+<a href="https://software.intel.com/node/1c3820b4-42a1-48ea-887a-2cf39a41ce53">recursive</a>
+   </td>
+   <td>Mrecursive
+   </td>
+   <td>N/A
```
[the remark this answers is not in the document]
Comment (psteinfeld) [reply documentation/OptionComparison.md]: Thanks!  I'll fix this.

**Comment by psteinfeld** [comment]
Thanks for all of your feedback.

-Mstack_arrays didn’t (and still doesn’t) seem directly related to Fortran language processing, so I didn’t include it.

Thanks for all of the rest of your feedback, though.  I just created a new pull request, but GitHub didn’t give me the option to add you as a reviewer.  Please let me know, either by email or a comment on GitHub if you approve.  I’ll wait to hear from you before merging.

Pete

**Pull request merged**